### ISSUE TITLE
[A11y]: Arrow key nav doesn't work in header overflow menu

### ISSUE THREAD

**Issue by warrenmblood** [carbon-design-system/carbon-ai-chat#1105]
### Browser

Firefox

### Operating system

MacOS

### Package version

v1.9.0-rc.0

### Automated testing tool and ruleset

Manual testing

### Description

<img width="666" height="313" alt="Image" src="https://github.com/user-attachments/assets/90e4258e-6150-48f8-83d6-be76da5bd725" />

### Reproduction / example

https://chat.carbondesignsystem.com/tag/next/demo/index.html

### Steps to reproduce

1. Go to [demo](https://chat.carbondesignsystem.com/tag/next/demo/index.html)
2. In Chat configuration deselect "Hide chat header" and select "Add menu options"
3. Open overflow menu in header
4. Note that arrow up/down doesn't work to navigate options

### Suggested severity

None

### Code of conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon-ai-chat/issues) for duplicate problems
